app/controllers/hq/partners_controller.rb

```diff
@@ -5,4 +5,11 @@ class Hq::PartnersController < ApplicationController
   def index
     @partners= Partner.all
   end
+
+  def show
+    if not (@partner= Partner.find_by_id(params[:id]))
```

Comment:
`Model.find` will raise an exception on invalid id, is that which method you refer to?